src/bloqade/lanes/layout/arch.py

```diff
@@ -36,6 +36,10 @@ class ArchSpec(Generic[SiteType]):
     """tuple of all words in the architecture. words[i] gives the word at word address i."""
     zones: tuple[tuple[int, ...], ...]
     """A tuple of zones where a zone is a tuple of word addresses and zone[i] gives the ith zone."""
+    measurement_zones: frozenset[int]
+    """Set of zone ids that support measurements."""
+    cz_gate_zones: frozenset[int]
```

Comment:
```suggestion
    entangling_zones: frozenset[int]
```
`ArcSpec` should be hardware agnostic and future architecture may implement non-cz gate.